packages/api/src/routers/article_router.ts

```diff
@@ -107,69 +114,30 @@ export function articleRouter() {
       if (existingSpeech?.state === SpeechState.INITIALIZED) {
         logger.info('Found existing in progress speech')
         // retry later
-        return res.status(429).send('Speech is in progress')
+        return res.status(202).send('Speech is in progress')
       }
 
-      logger.debug('Text to speech request', { articleId: id })
+      logger.info('Text to speech request', { articleId })
+      const page = await getPageById(articleId)
+      if (!page) {
+        return res.status(404).send('Page not found')
+      }
       const userPersonalization = await getRepository(
         UserPersonalization
       ).findOneBy({
         user: { id: uid },
       })
-      if (!userPersonalization) {
-        return res.status(404).send('User Personalization not found')
-      }
-
-      const page = await getPageById(id)
-      if (!page) {
-        return res.status(404).send('Page not found')
-      }
-
-      // const text = parseHTML(page.content).document.documentElement.innerText
-      // if (!text) {
-      //   return res.status(404).send('Page has no text')
-      // }
-
       // initialize state
       const speech = await getRepository(Speech).save({
         user: { id: uid },
-        elasticPageId: id,
+        elasticPageId: articleId,
         state: SpeechState.INITIALIZED,
-        voice: userPersonalization.speechVoice,
+        voice: voice || userPersonalization?.speechVoice,
       })
-      try {
-        const startTime = Date.now()
-        const speechOutput = await synthesizeTextToSpeech({
-          id,
-          text: page.content,
-          languageCode: page.language,
-          voice: userPersonalization.speechVoice,
-          textType: 'ssml',
-        })
-        logger.info('Created speech', {
-          audioFileName: speechOutput.audioFileName,
-          speechMarksFileName: speechOutput.speechMarksFileName,
-          duration: Date.now() - startTime,
-        })
-
-        // update state
-        await getRepository(Speech).update(speech.id, {
-          state: SpeechState.COMPLETED,
-          audioFileName: speechOutput.audioFileName,
-          speechMarksFileName: speechOutput.speechMarksFileName,
-        })
-        speech.audioFileName = speechOutput.audioFileName
-        speech.speechMarksFileName = speechOutput.speechMarksFileName
-
-        res.redirect(await redirectUrl(speech, outputFormat))
-      } catch (error) {
-        logger.error('Text to speech error', { error })
-        // update state
-        await getRepository(Speech).update(speech.id, {
-          state: SpeechState.FAILED,
-        })
-        res.status(500).send('Text to speech error')
-      }
+      // enqueue a task to convert text to speech
+      const taskName = await enqueueTextToSpeech(uid, speech.id)
```

Comment:
Making sure I understand: if the speech has been set to INITIALIZED, we return 202, if there is no speech, we create an initialized item and return 202 also. Right?

Reply:
Yes

Reply:
We return different response body